substra/runner.py

```diff
@@ -201,12 +201,48 @@ def compute_perf(pred_path, opener_file, fake_data_samples, data_path, docker_cl
     return perf['all']
 
 
+def raise_if_path_traversal(requested_paths, to_directory):
+    # Inspired from https://stackoverflow.com/a/45188896
+
+    safe_directory = os.path.join(
+        os.path.realpath(to_directory),
+        ''
+    )
+
+    if not isinstance(requested_paths, list):
+        raise TypeError(f'requested_paths argument should be a list not a {type(requested_paths)}')
+
+    for requested_path in requested_paths:
+        real_requested_path = os.path.realpath(requested_path)
+        path_traversal = os.path.commonprefix([real_requested_path,
+                                               safe_directory]) != safe_directory
```

Comment:
Does it prevent from something like this :
```python
safe_directory='/tmp/test/'
real_requested_path='/tmp/test/../../bin/file'
```

Because `os.path.commonprefix()` would return `/tmp/test/` as it is the longest matching prefix which is equal to `safe_directory`. But in the end the requested path still do path traversal and unzip to an unsafe directory.

Reply:
It should be ok thanks to the usage of `realpath`:

```
In [1]: os.path.realpath('/tmp/test/../../bin/file')
Out[1]: '/private/bin/file'
```

Reply:
@AlexandrePicosson, as Samuel said, 
`os.path.realpath` will resolve the `'../../'` to the realpath  (it also resolve simlinks)
So, in your example, it will raise an `Exception`. You can see the unit test in substra-backend to see that your example is handle :)